testing/tests/DevExpress.ui.widgets.treeList/editing.tests.js

```diff
@@ -286,6 +286,63 @@ QUnit.test('AddRow method should expand row and add item after parent', function
     assert.strictEqual(rows[2].node.parent.key, 1, 'row 2 node parent');
 });
 
+QUnit.test('AddRow method should return Deferred with collapsed parent', function(assert) {
+    // arrange
+    this.options.editing.allowAdding = true;
+
+    this.setupTreeList();
+    this.rowsView.render($('#treeList'));
+    this.clock.tick();
+
+    // assert
+    assert.equal(this.getVisibleRows().length, 1, 'one visible row');
+
+    // act
+    let doneExecuteCount = 0;
+    this.addRow(1).done(() => {
+        doneExecuteCount++;
+    });
+    this.clock.tick();
+
+    assert.equal(doneExecuteCount, 1, 'done was executed');
+    assert.equal(this.getVisibleRows().length, 3, 'parent was expanded and one more row was added');
+});
+
+QUnit.test('Sequential adding of a row after adding the previous using Deferred (T844118)', function(assert) {
+    // arrange
+    const initNewRowCalls = [];
+
+    this.options.editing.allowAdding = true;
+    this.options.onInitNewRow = (e) => {
+        initNewRowCalls.push(e.data.parentId);
+    };
+
+    this.setupTreeList();
+    this.rowsView.render($('#treeList'));
+    this.clock.tick();
+
+    // assert
+    assert.equal(this.getVisibleRows().length, 1, '1 visible row');
+
+    // act
+    const addRowAfterDeferredResolve = (parentIds, index) => {
+        const parentId = parentIds[index];
+
+        if(parentId !== undefined) {
+            return this.addRow(parentId).done(addRowAfterDeferredResolve.bind(null, parentIds, index + 1));
```

Comment:
Why is the bind method used here?

Reply:
Replaced with arrow function for more readability